### PR TITLE
db auth , role, roledao,roleservice classes implemented

### PR DESCRIPTION
**Pull request by semyonich**

### CODE REVIEW

File: src/main/java/com/cinema/controller/CinemaHallController.java [hunk 1]
```diff
@@ -31,7 +32,8 @@ public void addCinemaHall(@RequestBody @Valid CinemaHallRequestDto cinemaHallReq
     }
 
     @GetMapping
-    public List<CinemaHallResponseDto> getAllCinemaHalls() {
+    public List<CinemaHallResponseDto> getAllCinemaHalls(Authentication auth) {
+        System.out.println(auth);
```
Comment (mfedenko09) [inline src/main/java/com/cinema/controller/CinemaHallController.java]:
```suggestion
```

File: src/main/java/com/cinema/model/RoleName.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.cinema.model;
+
+public enum RoleName {
+    ROLE_USER,
+    ROLE_ADMIN
```
Comment (mfedenko09) [inline src/main/java/com/cinema/model/RoleName.java]:
```suggestion
    USER,
    ADMIN
```

File: src/main/java/com/cinema/security/impl/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -1,27 +1,38 @@
...
         this.shoppingCartService = shoppingCartService;
+        this.roleService = roleService;
     }
 
     @Override
     public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
+        Role userRole = roleService.getRoleByName("ROLE_USER");
```
Comment (mfedenko09) [inline src/main/java/com/cinema/security/impl/AuthenticationServiceImpl.java]:
```suggestion
        Role userRole = roleService.getRoleByName("USER");
```

File: src/main/java/com/cinema/util/InitialInject.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    public void inject() {
+        for (RoleName item : RoleName.values()) {
+            Role role = new Role();
+            role.setName(item);
+            roleService.add(role);
+        }
+        User admin = new User();
+        admin.setEmail(ADMIN_EMAIL);
+        admin.setPassword(ADMIN_PWD);
+        Role adminRole = roleService.getRoleByName(RoleName.ROLE_ADMIN.name());
```
Comment (mfedenko09) [inline src/main/java/com/cinema/util/InitialInject.java]:
```suggestion
        Role adminRole = roleService.getRoleByName(RoleName.ADMIN.name());
```

**Review by mfedenko09 — APPROVED** [review]

File: src/main/java/com/cinema/controller/CinemaHallController.java (hunk 1 above)
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/controller/CinemaHallController.java]: do we need sout here?

File: src/main/java/com/cinema/dao/impl/UserDaoImpl.java [hunk 5]
```diff
@@ -19,7 +19,8 @@ public UserDaoImpl(SessionFactory sessionFactory) {
     @Override
     public Optional<User> findByEmail(String email) {
         try (Session session = sessionFactory.openSession()) {
-            return session.createQuery("FROM User WHERE email=:email", User.class)
+            return session.createQuery("FROM User u "
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/dao/impl/UserDaoImpl.java:22]: use inner join here, there should not be a situation when you have user without roles

File: src/main/java/com/cinema/model/RoleName.java (hunk 2 above)
Comment (RomanDubovskyi) [reply src/main/java/com/cinema/model/RoleName.java]: \+

File: src/main/java/com/cinema/model/User.java [hunk 6]
```diff
@@ -16,6 +21,11 @@
     @Column(unique = true, length = 30)
     private String email;
     private String password;
+    @ManyToMany(fetch = FetchType.LAZY)
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/User.java]: it's lazy by default

File: src/main/java/com/cinema/security/CustomUserDetailsService.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String name) throws UsernameNotFoundException {
+        User user = userService.findByEmail(name)
+                .orElseThrow(() -> new DataProcessingException("User don't exist"));
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/security/CustomUserDetailsService.java]: throw UsernameNotFoundException here instead

File: src/main/java/com/cinema/util/InitialInject.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        List<Role> adminRoles = new ArrayList<>();
+        System.out.println(adminRole);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/util/InitialInject.java]: remove sout

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/controller/CinemaHallController.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/com/cinema/controller/CinemaHallController.java]: done

File: src/main/java/com/cinema/model/RoleName.java (hunk 2 above)
Comment (semyonich) [reply src/main/java/com/cinema/model/RoleName.java]: done

File: src/main/java/com/cinema/security/impl/AuthenticationServiceImpl.java (hunk 3 above)
Comment (semyonich) [reply src/main/java/com/cinema/security/impl/AuthenticationServiceImpl.java]: done

File: src/main/java/com/cinema/util/InitialInject.java (hunk 4 above)
Comment (semyonich) [reply src/main/java/com/cinema/util/InitialInject.java]: done

File: src/main/java/com/cinema/controller/CinemaHallController.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/com/cinema/controller/CinemaHallController.java]: done

File: src/main/java/com/cinema/dao/impl/UserDaoImpl.java (hunk 5 above)
Comment (semyonich) [reply src/main/java/com/cinema/dao/impl/UserDaoImpl.java:22]: done

File: src/main/java/com/cinema/model/User.java (hunk 6 above)
Comment (semyonich) [reply src/main/java/com/cinema/model/User.java]: done

File: src/main/java/com/cinema/util/InitialInject.java (hunk 8 above)
Comment (semyonich) [reply src/main/java/com/cinema/util/InitialInject.java]: done

File: src/main/java/com/cinema/security/CustomUserDetailsService.java (hunk 7 above)
Comment (semyonich) [reply src/main/java/com/cinema/security/CustomUserDetailsService.java]: done

File: src/main/java/com/cinema/security/CustomUserDetailsService.java [hunk 9]
```diff
@@ -23,11 +22,11 @@ public CustomUserDetailsService(UserService userService) {
     @Override
     public UserDetails loadUserByUsername(String name) throws UsernameNotFoundException {
         User user = userService.findByEmail(name)
-                .orElseThrow(() -> new DataProcessingException("User don't exist"));
+                .orElseThrow(() -> new UsernameNotFoundException("User don't exist"));
         org.springframework.security.core.userdetails.User.UserBuilder userBuilder =
                 org.springframework.security.core.userdetails.User.withUsername(user.getEmail());
         List<GrantedAuthority> authoritySet = user.getRoles().stream()
-                .map(r -> new SimpleGrantedAuthority(r.getName().name()))
+                .map(r -> new SimpleGrantedAuthority("ROLE_" + r.getName().name()))
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/security/CustomUserDetailsService.java:29]: do we need ROLE_ now ?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/security/CustomUserDetailsService.java (hunk 9 above)
Comment (semyonich) [reply src/main/java/com/cinema/security/CustomUserDetailsService.java:29]: SimpleGrantedAuthority needs roles with prefix, or app will not work correctly. Sofia said it is ok solution :)

Comment (RomanDubovskyi) [reply src/main/java/com/cinema/security/CustomUserDetailsService.java:29]: ok 👍

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**